tensorflow/python/ops/random_ops.py

```diff
@@ -40,6 +41,9 @@ def _ShapeTensor(shape):
   if isinstance(shape, (tuple, list)) and not shape:
     dtype = dtypes.int32
   else:
+    if isinstance(shape, (tuple, list)):
+      shape = [val.value if isinstance(
```

Comment:
use tensor_shape.dimension_value here instead of the explicit isinstance check (it does the isinstance check for you)

Reply:
Thanks @alextp. The PR has been updated.